src/main/scala/zio/prelude/AddMultiplyShape.scala

```diff
@@ -0,0 +1,100 @@
+package zio.prelude
+
+import zio.prelude.newtypes.{ Prod, Sum }
+
+trait AddMultiplyShape[A, +Addition[x] <: Associative[x], +Multiplication[x] <: Associative[x]] {
```

Comment:
This is the one I think should be called `Distributive`, because the fundamental lie here that binds the two binary operators is `a * (b + c) == a * b + a * c`. Then other laws add additional structure.

Reply:
Both `AnnihilatingZero` and `DistributiveMultiply` type classes share a common "shape". I introduced the trait `AddMultiplyShape` so that both type classes can inherit from it.